### ISSUE TITLE
Wrong index of label

### ISSUE THREAD

**Issue by PhamMinhHiepIT2** [abdurmasood/Irregular-Heartbeat-Detection-using-ML#16]
Current (2022-07-31) code: set labels when running function **trainAndTestSplit** is always {0,3}. It does not include index of classes ['N', 'R', 'V'].

```python
def trainAndTestSplit(df, size_of_test_data):
    '''
        take dataframe and divide it into train and 
    test data for model training
        Args:
                df (dataframe): dataframe with all images information
        images_to_train (int): number of images to get for training
                               from dataframe
        size_of_test_data (float): percentage of data specified for training
        Returns:
                X_train (list): list of training signals
        X_test (list): list of testing signals
        y_train (list): list of training classes
        y_test (list): list of testing classes
        '''
    image_count = 0
    classes_to_check = CLASSES_TO_CHECK
    images_available_in_class = IMAGES_TO_TRAIN

    # train + test data (signals and classes of signals respectively)
    X = []
    y = []

    for index, row in df.iterrows():
        # check if current row is one of the classes to classify
        if row['Type'] in classes_to_check:
            images_available_in_class = df['Type'].value_counts()[row['Type']]

            X.append(row['Signal'])
            y.append(classes_to_check.index(row['Type']))
            image_count += 1

            if images_available_in_class < IMAGES_TO_TRAIN:
                if image_count == df['Type'].value_counts()[row['Type']]:

                    image_count = 0
                    classes_to_check.remove(row['Type'])
            else:
                if image_count == IMAGES_TO_TRAIN:

                    image_count = 0
                    classes_to_check.remove(row['Type'])

        # if data collected from all classes break loop
        if len(classes_to_check) == 0:
            break

    # split x and y into train and test data
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=size_of_test_data)

    # convert to numpy array
    X_train, X_test, y_train, y_test = convertToNumpy(
        X_train, X_test, y_train, y_test)

    # normalize data for easy data processing
    X_train, X_test, y_train, y_test = normalizeData(
        X_train, X_test, y_train, y_test)

    return X_train, X_test, y_train, y_test
```

**Comment by PhamMinhHiepIT2** [comment]
> Current (2022-07-31) code: set labels when running function **trainAndTestSplit** is always {0,3}. It does not include index of classes ['N', 'R', 'V'].
> 
> ```python
> def trainAndTestSplit(df, size_of_test_data):
>     '''
>         take dataframe and divide it into train and 
>     test data for model training
>         Args:
>                 df (dataframe): dataframe with all images information
>         images_to_train (int): number of images to get for training
>                                from dataframe
>         size_of_test_data (float): percentage of data specified for training
>         Returns:
>                 X_train (list): list of training signals
>         X_test (list): list of testing signals
>         y_train (list): list of training classes
>         y_test (list): list of testing classes
>         '''
>     image_count = 0
>     classes_to_check = CLASSES_TO_CHECK
>     images_available_in_class = IMAGES_TO_TRAIN
> 
>     # train + test data (signals and classes of signals respectively)
>     X = []
>     y = []
> 
>     for index, row in df.iterrows():
>         # check if current row is one of the classes to classify
>         if row['Type'] in classes_to_check:
>             images_available_in_class = df['Type'].value_counts()[row['Type']]
> 
>             X.append(row['Signal'])
>             y.append(classes_to_check.index(row['Type']))
>             image_count += 1
> 
>             if images_available_in_class < IMAGES_TO_TRAIN:
>                 if image_count == df['Type'].value_counts()[row['Type']]:
> 
>                     image_count = 0
>                     classes_to_check.remove(row['Type'])
>             else:
>                 if image_count == IMAGES_TO_TRAIN:
> 
>                     image_count = 0
>                     classes_to_check.remove(row['Type'])
> 
>         # if data collected from all classes break loop
>         if len(classes_to_check) == 0:
>             break
> 
>     # split x and y into train and test data
>     X_train, X_test, y_train, y_test = train_test_split(
>         X, y, test_size=size_of_test_data)
> 
>     # convert to numpy array
>     X_train, X_test, y_train, y_test = convertToNumpy(
>         X_train, X_test, y_train, y_test)
> 
>     # normalize data for easy data processing
>     X_train, X_test, y_train, y_test = normalizeData(
>         X_train, X_test, y_train, y_test)
> 
>     return X_train, X_test, y_train, y_test
> ```

Because I cannot push my code into repo so I commented my solution to resolve this issue:

```python
def trainAndTestSplit(df, size_of_test_data):
    '''
        take dataframe and divide it into train and 
    test data for model training

        Args:
                df (dataframe): dataframe with all images information

        images_to_train (int): number of images to get for training
                               from dataframe

        size_of_test_data (float): percentage of data specified for training

        Returns:
                X_train (list): list of training signals

        X_test (list): list of testing signals

        y_train (list): list of training classes

        y_test (list): list of testing classes
        '''
    image_count = 0
    # avoid variable reference same address with constant CLASSES_TO_CHECK
    classes_to_check = CLASSES_TO_CHECK.copy()
    images_available_in_class = IMAGES_TO_TRAIN

    # train + test data (signals and classes of signals respectively)
    X = []
    y = []

    for index, row in df.iterrows():
        # check if current row is one of the classes to classify
        if row['Type'] in classes_to_check:
            images_available_in_class = df['Type'].value_counts()[row['Type']]

            X.append(row['Signal'])
            y.append(CLASSES_TO_CHECK.index(str(row['Type'])))

            image_count += 1

            if images_available_in_class < IMAGES_TO_TRAIN:
                if image_count == df['Type'].value_counts()[row['Type']]:

                    image_count = 0
                    classes_to_check.remove(row['Type'])
            else:
                if image_count == IMAGES_TO_TRAIN:

                    image_count = 0
                    classes_to_check.remove(row['Type'])

        # if data collected from all classes break loop
        if len(classes_to_check) == 0:
            break

    # split x and y into train and test data
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=size_of_test_data)

    # convert to numpy array
    X_train, X_test, y_train, y_test = convertToNumpy(
        X_train, X_test, y_train, y_test)

    # normalize data for easy data processing
    X_train, X_test, y_train, y_test = normalizeData(
        X_train, X_test, y_train, y_test)

    return X_train, X_test, y_train, y_test

```
